datafusion/core/src/physical_plan/sorts/sort.rs

```diff
@@ -238,6 +247,9 @@ impl ExternalSorter {
             return Ok(());
         }
 
+        // Release the memory reserved for merge
+        self.merge_reservation.free();
+
         self.in_mem_batches = self
             .in_mem_sort_stream(self.metrics.baseline.intermediate())?
```

Comment:
I double checked that `in_mem_sort_stream` correctly respects `self.reservation` 👍